Description: lazbuild fails to build some packages during auto tests.
Author: Abou Al Montacir <abou.almontacir@sfr.fr>
Bug-Debian: http://bugs.debian.org/943600
diff --git a/components/ideintf/ideexterntoolintf.pas b/components/ideintf/ideexterntoolintf.pas
index 4654265c..fdf7549f 100644
--- a/components/ideintf/ideexterntoolintf.pas
+++ b/components/ideintf/ideexterntoolintf.pas
@@ -1229,12 +1229,14 @@ end;
 
 procedure TAbstractExternalTool.EnterCriticalSection;
 begin
-  FWorkerMessages.EnterCriticalSection;
+  if Assigned(FWorkerMessages) then
+    FWorkerMessages.EnterCriticalSection;
 end;
 
 procedure TAbstractExternalTool.LeaveCriticalSection;
 begin
-  FWorkerMessages.LeaveCriticalSection;
+  if Assigned(FWorkerMessages) then
+    FWorkerMessages.LeaveCriticalSection;
 end;
 
 procedure TAbstractExternalTool.ConsistencyCheck;
